src/zeta/aca_zeta_programming.cpp

```diff
@@ -184,6 +220,83 @@ void start_upd_listener(uint oam_port_number)
   }
 }
 
+int ACA_Zeta_Programming::create_arion_config(const alcor::schema::GatewayConfiguration current_AuxGateway,
+                                              string subnet_cidr, uint tunnel_id){
+    ACA_LOG_DEBUG("%s", "ACA_Zeta_Programming::create_arion_config ---> Entering\n");
+    int overall_rc = EXIT_SUCCESS;
+    zeta_config *current_zeta_cfg;
+    bool bucket_not_found = false;
+    CTSL::HashMap<FWD_Info, int *, FWD_Info_Hash> new_zeta_buckets;
+
+    uint oam_port = current_AuxGateway.arion_info().port_inband_operation();
+
+    // -----critical section starts-----
+    _zeta_config_table_mutex.lock();
+    if (!_zeta_config_table.find(current_AuxGateway.id(), current_zeta_cfg)) {
+        create_entry(current_AuxGateway.id(), oam_port, current_AuxGateway);
+        _zeta_config_table.find(current_AuxGateway.id(), current_zeta_cfg);
+        overall_rc = _create_zeta_group_entry(current_zeta_cfg);
+
+        if (std::find(this->_listening_oam_ports.begin(), this->_listening_oam_ports.end(), oam_port) == this->_listening_oam_ports.end()){
+            ACA_LOG_INFO("Creating thread for port %d.\n", oam_port);
+            std::thread *oam_port_listener_thread = NULL;
+            // TODO: Need to track, and kill this thread when this zeta config is deleted.
+            oam_port_listener_thread = new std::thread(std::bind(&start_upd_listener, oam_port));
+            oam_port_listener_thread->detach();
+            ACA_LOG_INFO("Created thread for port %d and it is detached.\n", oam_port);
+        }else{
+            ACA_LOG_INFO("There already is a running thread for port %d, no need to create another one.\n", oam_port);
+        }
+    } else {
+        for (auto destination : current_AuxGateway.destinations()) {
+            FWD_Info target_fwd(destination.ip_address(), destination.mac_address());
+
+            int *not_used = nullptr;
+
+            if (current_zeta_cfg->zeta_buckets.find(target_fwd, not_used)) {
+                continue;
+            } else {
+                bucket_not_found |= true;
+                break;
+            }
+            new_zeta_buckets.insert(target_fwd, nullptr);
+        }
+
+        // If the buckets have changed, update the buckets and group table rules.
+        if (bucket_not_found == true) {
+            current_zeta_cfg->zeta_buckets.clear();
+            for (auto destination : current_AuxGateway.destinations()) {
+                FWD_Info target_fwd =
+                        FWD_Info(destination.ip_address(), destination.mac_address());
+
+                current_zeta_cfg->zeta_buckets.insert(target_fwd, nullptr);
+            }
+
+            overall_rc = _update_zeta_group_entry(current_zeta_cfg);
+        }
+    }
+    _zeta_config_table_mutex.unlock();
+    // -----critical section ends-----
+
+    // set punt for for this tunnel_id and subnet.
```

Comment:
'set punt for for this tunnel_id and subnet' is it done? Or no longer needed

Reply:
I put it to the right spot, which is above
```
_create_arion_group_punt_rule(tunnel_id, subnet_cidr, current_zeta_cfg->group_id);
```